File: python-pr/52/python/code-quality-readability-maintainability/log_parser_20.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def parse_log_file(log_file):
    import re
    with open(log_file) as f:
        lines = f.readlines()
    parsed = []
    for line in lines:
        match = re.match(r'(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) \[(\w+)\] (.+)', line)
        if match:
            timestamp, level, message = match.groups()
            parsed.append({'timestamp': timestamp, 'level': level, 'message': message})
    return parsed
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/log_parser_20.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function parseLogFile(logFile) {
    const fs = require('fs');
    let data = fs.readFileSync(logFile, 'utf8');
    let lines = data.split('\n');
    let parsed = [];
    for (let line of lines) {
        let match = line.match(/(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) \[(\w+)\] (.+)/);
        if (match) {
            let [, timestamp, level, message] = match;
            parsed.push({timestamp, level, message});
        }
    }
    return parsed;
}
// {/fact}